src/shared/services/json-store.service.spec.ts

```diff
@@ -86,4 +86,73 @@ describe('JsonStoreService', () => {
     service.setIn(path, value);
   });
 
+  it('should shift the first element if the new pushed element exceeds the maximum history length', () => {
+    let initialJson = fromJS({
+      aMap: {
+        anotherMap: {
+          innerMap: {
+            innerProperty: 'value'
+          }
+        }
+      }
+    });
+    service.setJson(initialJson);
+    service.addJsonToHistory();
+
+    let anotherJson = fromJS({
+      anotherMap: {
+        anotherAnotherMap: {
+          anotherInnerMap: {
+            anotherInnerProperty: 'anotherValue'
+          }
+        }
+      }
+    });
+    service.setJson(anotherJson);
+
```

Comment:
instead of hardcoding, `5`.
here 
`let historyLimit = service['historyLimit']`;